libr/core/cmd_open.c

```diff
@@ -782,35 +778,56 @@ static void cmd_open_map(RCore *core, const char *input) {
 	r_core_block_read (core);
 }
 
-R_API void r_core_file_reopen_in_malloc (RCore *core) {
-	RListIter *iter;
-	RList *files = r_id_storage_list (core->io->files);
-	RIODesc *desc;
-	r_list_foreach (files, iter, desc) {
-		if (strstr (desc->name, "://")) {
-			continue;
-		}
-		ut64 sz = r_io_desc_size (desc);
-		ut8 *buf = calloc (sz, 1);
-		if (!buf) {
-			eprintf ("Cannot allocate %d\n", (int)sz);
-			continue;
-		}
-		(void)r_io_pread_at (core->io, 0, buf, sz);
-		char *url = r_str_newf ("malloc://%d", (int)sz);
-		// use r_io_desc_exchange pls
-		RIODesc *newDesc = r_io_open (core->io, url, R_IO_READ | R_IO_WRITE, 0);
-		if (newDesc) {
-			r_io_desc_close (desc);
-			(void)r_io_write_at (core->io, 0, buf, sz);
-		} else {
-			eprintf ("Cannot open %s\n", url);
-		}
+static bool reopen_in_malloc_cb(void *user, void *data, ut32 id) {
+	RIO *io = (RIO *)user;
+	RIODesc *desc = (RIODesc *)data;
+	RIODesc *ndesc;
+	ut64 size;
+	ut8 *buf;
+	char *uri;
+
+	if (r_io_desc_is_blockdevice (desc) || r_io_desc_is_dbg (desc)) {
+		return true;
+	}
+
+	if (strstr (desc->uri, "://")) {
+		return true;
+	}
+
+	size = r_io_desc_size (desc);
+
+	uri = r_str_newf ("malloc://%"PFMT64u, size);
+	if (!uri) {
+		return false;
+	}
+
+	buf = malloc (size);
+// if malloc fails, we can just abort the loop by returning false
```

Comment:
The code self explains, no need for a comment.

Reply:
only for advanced programmers, like you